vendor/github.com/gobuffalo/envy/envy.go

```diff
@@ -3,27 +3,22 @@ package envy
 import (
 	"fmt"
 	"os"
-	"path/filepath"
+	"os/exec"
 	"runtime"
 	"strings"
 	"sync"
-
-	homedir "github.com/mitchellh/go-homedir"
 )
 
 var gil = &sync.Mutex{}
 var env = map[string]string{}
 
 func init() {
-	v := runtime.Version()
-	// set the GOPATH if using >= 1.8 and the GOPATH isn't set
-	if v >= "go1.8" && os.Getenv("GOPATH") == "" {
-		home, err := homedir.Dir()
+	// set the GOPATH if not present
+	if os.Getenv("GOPATH") == "" {
+		out, err := exec.Command("go", "env", "GOPATH").Output()
 		if err == nil {
-			home, err := homedir.Expand(home)
-			if err == nil {
-				os.Setenv("GOPATH", filepath.Join(home, "go"))
-			}
+			gp := strings.TrimSpace(string(out))
+			os.Setenv("GOPATH", gp)
```

Comment:
So, you propose to; for when the `GOPATH` environment variable isn't set, instead of setting it to `filePath.Join(homedir.Dir(), "go")` use the output of `go env GOPATH`. However if locally I run `go help env` I see:

> usage: go env [-json] [-u] [-w] [var ...]
>
> Env prints Go environment information.
>
> By default env prints information as a shell script
> (on Windows, a batch file). If one or more variable
> names is given as arguments, env prints the value of
> each named variable on its own line.

If I'm reading that correctly it means that `go env GOPATH` will print the value set for the `GOPATH`, and yet we are trying to set something that appears to be set ? . I'm confused 😕 . 

What the script is currently doing makes sense, if there is no `GOPATH` set already, set it to a "generally used" value; `<HOMEDIR>/go`. 



I don't see the need for removing the pre-existent condition `v >= "go1.8"`.

What am I missing ?

Reply:
> If I'm reading that correctly it means that go env GOPATH will print the value set for the GOPATH, and yet we are trying to set something that appears to be set ? 

Yes and no 😄 .  `go env GOPATH` will print the value set for the `GOPATH` on the GO environment, which is different from the OS environment.
So basically, if GOPATH is empty at OS environment level, we copy the value from the GO environment.

This is needed because [when we ask for the GoPaths](https://github.com/codeclimate/test-reporter/blob/master/formatters/gocov/gocov.go#L21), we [use the GOPATH at OS level](https://github.com/codeclimate/test-reporter/blob/master/vendor/github.com/gobuffalo/envy/envy.go#L110-L117).

> What the script is currently doing makes sense, if there is no GOPATH set already, set it to a "generally used" value; <HOMEDIR>/go.

Yep, but that didn't aged well, you can see the source code in the `envy` library, and they introduced commits like this one https://github.com/gobuffalo/envy/commit/c823918c9f8da35fa1ff645fc9b82a3611c86c8e. You can see that in [their current release](https://github.com/gobuffalo/envy/blob/main/envy.go#L51-L58) their `GOPATH` detection is the same code I pasted here. 

> I don't see the need for removing the pre-existent condition v >= "go1.8".

This condition was also removed in the original source code, and also, in our case, it will be always true because we're using `go1.15`

Reply:
oh, that calms me a lot. I'll settle if you explain me why monkey patch `envy` instead of updating it ?